department-of-veterans-affairs/vets-website
# Pension 68553 mvp step 1 applicant info add va claim history

## ⚠️ CAUTION: DO NOT MERGE THIS PR UNTIL 'PENSION_FORM_ENABLED' IS DISABLED IN PRODUCTION

## Summary
On the Pension Benefits application(21P-527EZ),
On the Step 1 Applicant Information page

Add new form field:
'Have you filed any type of VA claim before?' ( Yes ) ( No )

And a VA file number field when Yes is selected

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`
4. Go to `http://localhost:3001/pension/application/527EZ` in your browser

## How to verify
1. Select Start application or continue application
2. On Step 1 - Applicant Information verify the new field is added
3. Verify the 'VA file number' field is only visible when Yes is selected
5. Verify the 'VA file number' has a validation check
6. Verify changes are saved to the redux store and submitted to the save endpoint

## Related issue(s)
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68553)
[Epick in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots
_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

### Mobile
| Before | After | Yes | No | Error | Valid |
| ------ | ----- | ----- | ----- | ----- | ----- |
|![pension-app-prefilled-old(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/189830d1-cfc2-4250-9396-05024613d006)|![pension-app-prefilled(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/8fb23cf4-191c-4c1e-beb5-a8ea4fe6e0ca)|![pension-app-va-file-claimed-yes(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/0ef5a8fe-be62-4ee0-93dd-4847f6d4ea13)|![pension-app-va-file-claimed-no(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/c5330ac0-b434-433d-a6cb-fb13091e82cd)|![pension-app-va-file-number-error(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/c02aa091-e95b-4929-a94a-e76cab4cbe59)|![pension-app-va-file-number-valid(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/34fe730e-4762-4f9f-acc8-838192f55e53)

### Desktop
| Before | After | Yes | No | Error | Valid |
| ------ | ----- | ----- | ----- | ----- | ----- |
|![pension-app-prefilled-old](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/6667c1b3-a9a3-41f4-ac7a-155d078763be)|![pension-app-prefilled](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/6ae42bd4-5275-4469-9984-da0c4cee6034)|![pension-app-va-file-claimed-yes](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/0723c3b7-409c-4304-9377-2c694d129621)|![pension-app-va-file-claimed-no](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/0a89f317-a1c6-40b7-ba2e-013eb668ff0f)|![pension-app-va-file-number-error](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/7efb8055-f409-464c-a426-3cf5cb8737f3)|![pension-app-va-file-number-valid](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/d0153ff9-915c-44ab-8ff2-43f811e61e59)

## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
- On the application forms fist chapter and page, there will be a new field in between the Social Security Number field and the Date of Birth field
- It will be labeled “Have you filed any type of VA claim before?” with options for “Yes” and “No.”
- If the user selects “Yes,” an additional field labeled “VA File Number” will appear.
- The form will allow for a Veteran whose file number is different than their SSN to enter both. This field will be optional for the user.
- The “VA File Number” field should have validation to ensure it follows the correct format.
- The field responses to this question should be saved so that it can be submitted.

### Quality Assurance & Testing

### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
